configs/components/openssl-1.0.2.rb

```diff
@@ -84,6 +84,7 @@
     # we can ensure everything is build with the same compiler.
     pkg.apply_patch 'resources/patches/openssl/openssl-mingw-do-not-build-applink.patch'
     pkg.apply_patch 'resources/patches/openssl/openssl-1.0.2t-fix-no-asm-build-on-windows.patch'
+    pkg.apply_patch 'resources/patches/openssl/openssl-enforce-fips-on-fips-mode.patch'
```

Comment:
I think we should apply this only for windowsfips, right?

Reply:
changed code is inside #ifdef OPENSSL_FIPS, should be safe to apply it to all fips openssl builds (https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/o_init.c#L79)